lib/db_connection/connection.ex

```diff
@@ -381,4 +376,29 @@ defmodule DBConnection.Connection do
         [{mod, fun, length(args), info} | rest]
     end
   end
+
+  defp handle_ping(state, s) do
+    %{mod: mod, ping_timeout: timeout} = s
+
+    task =
+      Task.async(fn ->
```

Comment:
I think we should use the task pool because it handles test cases and ensures semantics. When the task pool was introduced we needed a non-blocking interface but I think we can simplify things greatly by moving that pool to a similar async/yield model. Lets use a supervisor and `async_nolink` though. Perhaps refactor after connect to that model first, then look at using it for ping. Any functions should be inside a DBConnection.run call if outside the connection process, and we need to be using `Holder` too.